components/function-controller/design/extend_function_spec_proposal.md

```diff
@@ -0,0 +1,392 @@
+  # metadata
+  buildSpec:
+    # resources
+    # envs (?)
+    # metadata (?)
+```
+
+### Mounts - own structure or k8s inherited
+
+Under the hood the secret mount becomes a volume mount in the runtime pod.
+We could :
+ - expose the k8s volume mount spec  in function spec
+ - cover k8s volume mount spec via oor own [specialised](https://servicebinding.io/application-developer/) spec
+
+```yaml 
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+spec:
+  volumeMounts:
+  - name: foo
+    mountPath: "/etc/foo"
+    readOnly: true
+  volumes:
+  - name: foo
+    secret:
+      secretName: mysecret
+```
+
+```yaml 
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+spec:
+  serviceBindings:
+  - source: my-secret
+  env:
+  - name: SERVICE_BINDING_ROOT 
+    value: /foo
+```
+
+### Metadata for function pod
+
+Define runtime labels and annotations on the root level or under a `metadata` field
+
+```yaml
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    ...
+  annotations:
+    ...
+spec:
+  metadata: 
+    labels: ...
+    annotations: ...
+```
+OR
+```yaml
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    ...
+  annotations:
+    ...
+spec:
+  labels: ...
+  annotations: ...
+```
+
+## Samples: 
+
+### Option 1
+
+- simplified mounts serving just for service binding purpose
+- extract `build` object for any build-time specific config
+- runtime labels and annotations on the spec root level
+
+```yaml
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    app.kubernetes.io/name: my-function
+spec:
+  runtime: nodejs16
+  source:
+    ...
+  serviceBindings: 
+  - source: my-secret
+    mountPath: "/foo" # optional.. read from SERVICE_BINDING_ROOT ENV 
+
+  profile: S / M / L / XL / ... / Custom #optional
+  resources: # optional... required if spec.profile==Custom
+    limits: ... #if_custom
+    requests: ... #if_custom
+​
+  
+  labels: 
+    app: my-app
+  annotations: 
+    fluentbit.io/parser: my-regex-parser
+    istio-injection: enabled
+​
+  env:
+  - name: SERVICE_BINDING_ROOT
+    value: /service_bindings
+​
+​
+  build:
+    profile: S / M / L / XL / ... / Custom
+    resources: # optional... required if spec.build.profile==Custom
+      limits: ... #if_custom
+      requests: ... #if_custom
+​
+```
+### Option 2
+
+- almost the same as `Option 1` (same pros)
+- dif1: move some fields from `.spec` to the new struct `.spec.runtimeSpec` to clearly distinguish fields desired to be used in the `build` and `running` phases. For example in `Option 1` users may have questions after seeing `.spec.env` and `.spec.build.env` fields for example "is .spec.env dedicated for the running function's pod? Would the field be merged with .spec.build.env for the building job?"
```

Comment:
Also agree with this. The split is confusing especially considering our current API.

Reply:
included in the last "compromise option"